### ISSUE TITLE
Translation of validation messages

### ISSUE THREAD

**Issue by maximehuran** [magento/magento2#8179]
### Preconditions

Magento Version is 2.1.3

### Steps to reproduce
1. Call a validation message in lib/web/mage/validation.js

### Expected result

1. Have a translated string

### Actual result

1. No translation

I change : 
```
"validate-email": [
            function (v) {
                return $.mage.isEmptyNoTrim(v) || /^([a-z0-9,!\#\$%&'\*\+\/=\?\^_`\{\|\}~-]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+(\.([a-z0-9,!\#\$%&'\*\+\/=\?\^_`\{\|\}~-]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+)*@([a-z0-9-]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+(\.([a-z0-9-]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+)*\.(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]){2,})$/i.test(v);
            },
            'Please enter a valid email address (Ex: johndoe@domain.com).'
        ],
```

To : 
```
"validate-email": [
            function (v) {
                return $.mage.isEmptyNoTrim(v) || /^([a-z0-9,!\#\$%&'\*\+\/=\?\^_`\{\|\}~-]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+(\.([a-z0-9,!\#\$%&'\*\+\/=\?\^_`\{\|\}~-]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+)*@([a-z0-9-]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+(\.([a-z0-9-]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+)*\.(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]){2,})$/i.test(v);
            },
            $.mage.__('Please enter a valid email address (Ex: johndoe@domain.com).')
        ],
```

My modification is : 
**$.mage.__('Please enter a valid email address (Ex: johndoe@domain.com).')**

And it works !

**Comment by veloraven** [comment]
@maximehuran thank you for your report.
Please, also identify which version of Magento you are running.

**Comment by maximehuran** [comment]
Magento Version is 2.1.3

Le mer. 18 janv. 2017 à 16:20, Olga Moyseyenko <notifications@github.com> a
écrit :

**Comment by slackerzz** [comment]
did you add the translation to your theme or in a module?
I've found that if you put translation in a theme they doesn't work as expected, move them to a module.
maybe it's related to https://github.com/magento/magento2/issues/6563

**Comment by quienti** [comment]
Already fixed in develop branch => https://github.com/magento/magento2/blob/develop/lib/web/mage/validation.js#L496

Commit => https://github.com/magento/magento2/commit/3109c8e96bcd08041f558be65a2a4e3d213d22a4

So, probably fixed in Magento 2.2

**Comment by KrystynaKabannyk** [comment]
Internal ticket MAGETWO-61566

**Comment by Triloworld** [comment]
Thanks for that fix. I think is incomplete. Better approach:

Function in `validation.js`: 
`$.validator.addMethod`

add this line:
`$.validator.messages[name] = $.mage.__($.validator.messages[name]);`

It will by default translate all built-in and new validation method.

**Comment by magento-engcom-team** [comment]
@maximehuran, thank you for your report.
The issue is already fixed in 2.2.0
